File: verify_neuro.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage();

  // Navigate to the local server
  try {
    await page.goto('http://localhost:3000/neuro.html');

    // Wait for the app to initialize
    await page.waitForTimeout(2000);

    // Click the "Start Simulation" button if it exists (it's in an overlay)
    const startBtn = await page.$('#neuro-start-overlay button');
    if (startBtn) {
        await startBtn.click();
    }

    // Wait for some simulation to happen
    await page.waitForTimeout(3000);

    // Take a screenshot
    await page.screenshot({ path: '/home/jules/verification/neuro_check.png' });
    console.log('Neuro page screenshot saved to /home/jules/verification/neuro_check.png');

    // Check for errors in the console
    page.on('console', msg => {
        if (msg.type() === 'error') {
            console.log(`PAGE ERROR: ${msg.text()}`);
        }
    });

  } catch (err) {
    console.error('Error during verification:', err);
  } finally {
    await browser.close();
  }
})();

File: verify_neuro.py
import asyncio
from playwright.async_api import async_playwright
import os

async def run():
    async with async_playwright() as p:
        browser = await p.chromium.launch()
        page = await browser.new_page()

        # Get absolute path
        abs_path = os.path.abspath('docs/test_models.html')
        url = f'file://{abs_path}'
        print(f"Navigating to {url}")

        await page.goto(url)

        # Wait for the page to load
        await page.wait_for_timeout(3000)

        # Select "neuro" from the dropdown
        print("Selecting Neuro Model from dropdown")
        await page.select_option('#model-selector', 'neuro')

        # Wait for the model to load and render
        print("Waiting for model to render...")
        await page.wait_for_timeout(8000)

        # Ensure verification dir exists
        os.makedirs('/home/jules/verification', exist_ok=True)

        screenshot_path = '/home/jules/verification/neuro_brain_realistic_v4.png'
        await page.screenshot(path=screenshot_path)
        print(f"Neuro Model screenshot captured at {screenshot_path}")

        await browser.close()

if __name__ == '__main__':
    asyncio.run(run())
